pkg/inject/virtualgateway_envoy.go

```diff
@@ -108,21 +94,72 @@ func (m *virtualGatewayEnvoyConfig) mutate(pod *corev1.Pod) error {
 	for name, value := range newEnvMap {
 		e := corev1.EnvVar{Name: name,
 			Value: value}
-		pod.Spec.Containers[envoyIdx].Env = append(pod.Spec.Containers[envoyIdx].Env, e)
+		envoy.Env = append(envoy.Env, e)
 	}
 
 	// customer can bring their own envoy image/spec for virtual gateway so we will only set readiness probe if not already set
-	if pod.Spec.Containers[envoyIdx].ReadinessProbe == nil {
-		pod.Spec.Containers[envoyIdx].ReadinessProbe = envoyReadinessProbe(m.mutatorConfig.readinessProbeInitialDelay,
+	if envoy.ReadinessProbe == nil {
+		envoy.ReadinessProbe = envoyReadinessProbe(m.mutatorConfig.readinessProbeInitialDelay,
 			m.mutatorConfig.readinessProbePeriod, strconv.Itoa(int(m.mutatorConfig.adminAccessPort)))
 	}
 
 	if m.mutatorConfig.enableSDS && !isSDSDisabled(pod) {
-		mutateSDSMounts(pod, &pod.Spec.Containers[envoyIdx], m.mutatorConfig.sdsUdsPath)
+		mutateSDSMounts(pod, &envoy, m.mutatorConfig.sdsUdsPath)
 	}
+	pod.Spec.Containers[envoyIdx] = envoy
 	return nil
 }
 
+func (m *virtualGatewayEnvoyConfig) getEnvMapForVirtualGatewayEnvoy(vars VirtualGatewayEnvoyVariables) map[string]string {
```

Comment:
Let's enhance the existing `buildSidecar` function to handle VG scenario as well instead of creating a separate function.